File: csw-location/src/main/scala/csw/services/location/javadsl/JActorRuntime.scala
package csw.services.location.javadsl

import java.util

import csw.services.location.common.ActorRuntime
import csw.services.location.models.IActorRuntime
import scala.collection.JavaConverters._

object JActorRuntime extends IActorRuntime{
  override def create(name: String, settings: util.Map[String, AnyRef]): ActorRuntime = new ActorRuntime(name, settings.asScala.toMap)

  override def create(name: String): ActorRuntime = new ActorRuntime(name)
}

File: csw-location/src/main/java/csw/services/location/javadsl/JActorRuntime.java
package csw.services.location.javadsl;

import csw.services.location.scaladsl.ActorRuntime;
import csw.services.location.common.ScalaCompat;

import java.util.Collections;
import java.util.Map;

public class JActorRuntime {

    public static ActorRuntime actorRuntime(String name, Map<String, Object> settings) {
        return new ActorRuntime(name, ScalaCompat.toMap(settings));
    }

    public static ActorRuntime actorRuntime(String name) {
        return actorRuntime(name, Collections.emptyMap());
    }

}
